### ISSUE TITLE
C# client, also get JSON string

### ISSUE THREAD

**Issue by stigc** [RSuter/NSwag#1345]
Is there a way to also get the JSON string besides the deserialized object?

**Comment by stigc** [comment]
I cannot see how this would be possible. Would a change like this be a solution for this?

https://github.com/stigc/NSwag/commit/8a7d0bf4f9bff0282b2832dc2859ca245022098a

**Comment by stigc** [comment]
Or maybe ProcessResponse(client_, response_); should be move to after the content has been buffered as a string and then responseData_ should be provided as well.
 
ProcessResponse(client_, response_, responseData_)

**Comment by RSuter** [comment]
There is no easy way to do this (e.g. no setting) but maybe you can do this by overwriting templates and stripping away lots of features: 

https://github.com/RSuter/NSwag/wiki/Templates

**Comment by stigc** [comment]
IMO it should be possible out of the box. The purpose is for logging where it would be nice to have the original JSON,

**Comment by stigc** [comment]
Yes I still need the POCOs and then have the plain JSON for logging.

Moving the partial function call ProcessResponse to the point where the client has the JSON string makes sense to me.

**Comment by RSuter** [comment]
Ref: https://github.com/RSuter/NSwag/issues/1333

Would be good if we also support base class

**Comment by stigc** [comment]
Yes or just make it as callbacks on an interface. To start with could we make a ProcessResponse partial function after the JSON string is buffered?
